extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -15,4 +32,37 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols() {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true, GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void registerVertxGraphqlUI(VertxGraphqlRecorder recorder,
+            BuildProducer<NativeImagePackageResourceBuildItem> nativeResourcesProducer, VertxGraphqlConfig config,
+            LaunchModeBuildItem launchMode, BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints,
+            BuildProducer<RouteBuildItem> routes) {
+        boolean includeVertxGraphqlUi = launchMode.getLaunchMode().isDevOrTest() || config.ui.alwaysInclude;
+
+        if (!includeVertxGraphqlUi) {
+            return;
+        }
+
+        String path = config.ui.path.replaceAll("/+$", "");
+        if (path.isEmpty()) {
+            throw new ConfigurationError(
```

Comment:
I think we should use `ConfigurationException` now.

Reply:
done